j2k/src/org/jetbrains/kotlin/j2k/ExpressionConverter.kt

```diff
@@ -501,6 +501,18 @@ class DefaultExpressionConverter : JavaElementVisitor(), ExpressionConverter {
                 }
             }
 
+            if (type?.canonicalText == CommonClassNames.JAVA_LANG_STRING) {
+                val argument = expression.argumentList?.expressions?.singleOrNull()
+                if (argument != null && argument.type?.canonicalText == CommonClassNames.JAVA_LANG_STRING) {
+                    result = if ((argument.reference?.resolve() as? PsiVariable)?.hasModifierProperty(PsiModifier.FINAL) == true) {
```

Comment:
This is not what I meant as "non-constant strings". Simply check whether the argument is a `PsiLiteralExpression`.

Reply:
Fixed